test/README.md

```diff
@@ -1,84 +0,0 @@
-# Bristlemouth Common Unit Tests
```

Comment:
[Two weeks ago we agreed](https://github.com/bristlemouth/bm_core/pull/22#discussion_r1815555631) to keep this file here and reference it in the API docs. Did something change that I haven't heard about?

Here are some of my thoughts/assumptions right now. Let's discuss! Happy to change my mind once we're on the same page. 😄 

1. I think curious developers, exploring Bristlemouth repos on github, will stumble onto a readme in the test folder more easily and fluidly than they will the documentation. In my own experience I often click around interesting github repos for projects I'm vaguely curious about, but I rarely go to their documentation until I'm definitely working on a related task.
2. This readme is targeted to folks actually trying to build and run the tests. It's the kind of thing, if I were just idly curious and exploring I'd be psyched that the project had unit tests, and I'd be curious how easy it was to run them, and how much coverage they had.
3. The API docs have a mix of intended audiences. Some of the docs are very general, architecture diagrams, helping people build an overall mental model of how Bristlmouth firmware works and what it can do. OTOH some of the docs are very targeted to a specific module or development task, like the migration guides we were talking about, for folks already deep in the code trying to solve a specific problem.
4. It helps with maintainability for us to have one place with this content rather than duplicating it and letting one side potentially get out of date.

So given that, I don't have a strong opinion about which side contains the content. I'm open to either. I do, however, have a strong opinion that both sides should exist, and one should link to the other. I specifically want to hear each of you @matt001k @victorsowa12 share your thoughts here in a reply. What do you think? Which place would you prefer? Any examples of other projects that you like and would want to emulate regarding testing instructions?

Reply:
Ahhhh! I forgot all about this conversation (there has been a lot of developments in the last two weeks 😄 !).

I think that having the integration docs point with a link to the `test/README.md` file in the main branch of `bm_core` would be a great move. That way the docs don't have multiple sources of truth and people looking to contribute can easily navigate to it in multiple locations (integration docs or the repository itself).

I do think that the test/README.md should contain the content as it is easier to navigate to when creating unit tests and seems to flow a bit better.